truffle/src/main/java/org/truffleruby/core/format/convert/Integer16BigToBytesNode.java

```diff
@@ -21,6 +21,12 @@
 
     @Specialization
     public byte[] encode(long value) {
+        if(value == 0x7fc00000 || value == 0x7ff8000000000000L){
+            return new byte[]{
+                (byte) 0x000000c0,
+                (byte) 0x000000ff
+            };
+        }
```

Comment:
This is not good as it might affect other types than doubles.
If something needs to change it should be in `ReinterpretAsLongNode`.
Maybe try `doubleToRawLongBits`.